File: String/165. Compare Version Numbers/compareVersionNumbers.js
/**
 * Time: O(n) | Space: O(1)
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
const compareVersion = function(version1, version2) {
  const v1Arr = version1.split(".");
  const v2Arr = version2.split(".");

  const max = Math.max(v1Arr.length, v2Arr.length);

  for (let i = 0; i < max; i++) {
    const v1 = parseInt(v1Arr[i]) || 0;
    const v2 = parseInt(v2Arr[i]) || 0;
    if (v1 > v2) {
      return 1;
    } else if (v1 < v2) {
      return -1;
    } else {
      continue;
    }
  }

  return 0;
};

File: String/165. Compare Version Numbers/compare_version_numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1_arr = version1.split('.')
        v2_arr = version2.split('.')

        max_length = max(len(v1_arr), len(v2_arr))

        for i in range(max_length):
            v1 = int(v1_arr[i]) if i < len(v1_arr) else 0
            print(type(v1))
            v2 = int(v2_arr[i]) if i < len(v2_arr) else 0
            print(type(v2))

            if v1 > v2:
                return 1
            elif v1 < v2:
                return -1
            else:
                continue

        return 0